### PR TITLE
RavenDB-12735 catch NotSupportedException and rethrow as IOException

### PR DESCRIPTION
**Pull request by karmeli87**
remove this workaround after updating to .net core 3
https://github.com/dotnet/corefx/issues/36141

### CODE REVIEW

File: src/Sparrow/Json/BlittableJsonTextWriter.cs [hunk 1]
```diff
@@ -755,6 +755,17 @@ public virtual void Dispose()
             {
                 //we are disposing, so this exception doesn't matter
             }
+            // TODO: remove when we update to .net core 3
+            // https://github.com/dotnet/corefx/issues/36141
+            catch (NotSupportedException e)
+            {
+                if (e.Message.Contains("The WriteAsync method cannot be called when another write operation is pending"))
```
Comment (ayende) [inline src/Sparrow/Json/BlittableJsonTextWriter.cs]: Not a good idea, may get different values under different cultures. 
Just wrap the exception

Comment (karmeli87) [reply src/Sparrow/Json/BlittableJsonTextWriter.cs]: Okay, but what if we getting a different `NotSupportedException`? what about to check only for the word `WriteAsync` ?

Comment (ayende) [reply src/Sparrow/Json/BlittableJsonTextWriter.cs]: I don't consider this likely, and the idea is that we'll restart anyway

Comment (karmeli87) [reply src/Sparrow/Json/BlittableJsonTextWriter.cs]: okay, done.

**Review by ppekrol — APPROVED** [review]

**Pull request merged**